internal/cmdrender/executor.go

```diff
@@ -398,7 +418,7 @@ func (pn *pkgNode) runValidators(ctx context.Context, hctx *hydrationContext, in
 		}
 		// validators are run on a copy of mutated resources to ensure
 		// resources are not mutated.
-		if _, err = validator.Filter(cloneResources(input)); err != nil {
+		if _, err = validator.Filter(selectInput(cloneResources(input), fn.Selectors)); err != nil {
```

Comment:
`cloneResources(selectInput(input))` should be faster.

Reply:
Done.